USAGE.md

```diff
@@ -0,0 +1,260 @@
+In case this is not set, the SDK will use the value of the `OP_VAULT` environment variable as the default UUID.
+If a field is within a section, the `opsection` tag is required as well. Please note that one cannot retrieve a section in itself.
+
+### Example Struct
+
+This example struct will retrieve 3 fields from one item and a whole item from another vault:
+
+```go
+package main
+
+import (
+	"github.com/1Password/connect-sdk-go/connect"
+	"github.com/1Password/connect-sdk-go/onepassword"
+)
+
+type Config struct {
+	Username string           `opitem:"Demo TF Database" opfield:"username"`
+	Password string           `opitem:"Demo TF Database" opfield:"password"`
+    Host     string           `opitem:"Demo TF Database" opsection:"details" opfield:"hostname"`
+	APIKey   onepassword.Item `opvault:"7vs66j55o6md5btwcph272mva4" opitem:"API Key"`
+}
+
+func main() {
+	client, err := connect.NewClientFromEnvironment()
+	if err != nil {
+		panic(err)
+	}
+    	c := Config{}
+	err = client.LoadStruct(&c)
+}
+
+```
+
+Additionally, fields of the same item can be added to a struct at once, without needing to specify the `opitem` or `opvault` tags:
+
+```go
+package main
+
+import "github.com/1Password/connect-sdk-go/connect"
+
+type Config struct {
+	Username string     `opfield:"username"`
+	Password string     `opfield:"password"`
+}
+
+func main () {
+	client, err := connect.NewClientFromEnvironment()
+    	if err != nil {
+		panic(err)
+	}
+	c := Config{}
+
+    // retrieve using item title
+	err = client.LoadStructFromItemByTitle(&c, "Demo TF Database", "7vs66j55o6md5btwcph272mva4")
+
+    // retrieve using item uuid
+    err = client.LoadStructFromItem(&c, "4bc73kao58g2usb582ndn3w4", "7vs66j55o6md5btwcph272mva4")
+}
+```
+
+## Environment Variables
+
+The Connect Go SDK makes use of the following environment variables:
+
+- `OP_CONNECT_TOKEN`: the API token to be used to authenticate the client to your 1Password Connect instance. Used in order to authenticate via the `connect.NewClientFromEnvironment` function.
+- `OP_CONNECT_HOST`: the hostname of your 1Password Connect instance. Used in order to authenticate via the `connect.NewClientFromEnvironment` function.
+- `OP_VAULT`: a vault UUID. Used as default vault in the `LoadStruct`, `LoadStructFromItemByTitle` and `LoadStructFromItem` functions, for all fields where the `opvault` tag is not set.
+
+## Errors
+
+All errors returned by Connect API are unmarshalled into a `onepassword.Error` struct:
+
+```go
+type Error struct {
+    StatusCode int    `json:"status"`
+    Message    string `json:"message"`
+}
+```
+
+Details of the errors can be accessed by using `errors.As()`:
+
+```go
+_, err := client.GetVaults()
+if err != nil{
+    var opErr *onepassword.Error
+    if errors.As(err, &opErr){
+        fmt.Printf("message=%s, status code=%d\n",
+            opErr.Message,
+            opErr.StatusCode,
+        )
+    }
+}
+```
```

Comment:
This codeblock uses 4 spaces for indentation, but all of the previous codeblocks use 2 spaces. I'm assuming that this one should be updated to use 2 as well?

Reply:
The issue is that previous snippets used the tab character (which can usually be mistaken for 4 spaces in an editor) compared to 4 spaces. This should be fixed now.